src/Diagram.js

```diff
@@ -248,17 +252,23 @@ class Diagram extends React.PureComponent {
     this.updateScroll(e.currentTarget);
   };
 
-  scrollToPosition = pos => {
-    this.updateScroll(pos);
-    this.containerRef.current.scrollTo(pos.scrollLeft, pos.scrollTop);
+  setZoom = newZoom => this.setState(prevState => ({ ...prevState, zoom: newZoom }));
```

Comment:
Why do we need to maintain this in current state, can this be been done inside contextProvider ?

Reply:
Yes multiple consumers in the diagram Class